test/e2e/network/ingress.go

```diff
@@ -935,3 +939,220 @@ func detectNegAnnotation(f *framework.Framework, jig *e2eingress.TestJig, gceCon
+		}
+		// Discovery
+		ginkgo.By("getting /apis")
+		{
+			discoveryGroups, err := f.ClientSet.Discovery().ServerGroups()
+			framework.ExpectNoError(err)
+			found := false
+			for _, group := range discoveryGroups.Groups {
+				if group.Name == networkingv1beta1.GroupName {
+					for _, version := range group.Versions {
+						if version.Version == ingVersion {
+							found = true
+							break
+						}
+					}
+				}
+			}
+			framework.ExpectEqual(found, true, fmt.Sprintf("expected networking API group/version, got %#v", discoveryGroups.Groups))
+		}
+
+		ginkgo.By("getting /apis/networking.k8s.io")
+		{
+			group := &metav1.APIGroup{}
+			err := f.ClientSet.Discovery().RESTClient().Get().AbsPath("/apis/networking.k8s.io").Do(context.TODO()).Into(group)
+			framework.ExpectNoError(err)
+			found := false
+			for _, version := range group.Versions {
+				if version.Version == ingVersion {
+					found = true
+					break
+				}
+			}
+			framework.ExpectEqual(found, true, fmt.Sprintf("expected networking API version, got %#v", group.Versions))
+		}
+
+		ginkgo.By("getting /apis/networking.k8s.io" + ingVersion)
+		{
+			resources, err := f.ClientSet.Discovery().ServerResourcesForGroupVersion(networkingv1beta1.SchemeGroupVersion.String())
+			framework.ExpectNoError(err)
+			foundIngress := false
+			for _, resource := range resources.APIResources {
+				switch resource.Name {
+				case "ingresses":
+					foundIngress = true
+				}
+			}
+			framework.ExpectEqual(foundIngress, true, fmt.Sprintf("expected ingresses, got %#v", resources.APIResources))
+		}
+
+		// Ingress resource create/read/update/watch verbs
+		ginkgo.By("creating")
+		_, err := ingClient.Create(context.TODO(), ingTemplate, metav1.CreateOptions{})
+		framework.ExpectNoError(err)
+		_, err = ingClient.Create(context.TODO(), ingTemplate, metav1.CreateOptions{})
+		framework.ExpectNoError(err)
+		createdIngress, err := ingClient.Create(context.TODO(), ingTemplate, metav1.CreateOptions{})
+		framework.ExpectNoError(err)
+
+		ginkgo.By("getting")
+		gottenIngress, err := ingClient.Get(context.TODO(), createdIngress.Name, metav1.GetOptions{})
+		framework.ExpectNoError(err)
+		framework.ExpectEqual(gottenIngress.UID, createdIngress.UID)
+
+		ginkgo.By("listing")
+		ings, err := ingClient.List(context.TODO(), metav1.ListOptions{FieldSelector: "metadata.namespace=" + ns})
+		framework.ExpectNoError(err)
+		framework.ExpectEqual(len(ings.Items), 3, "filtered list should have 3 items")
+
+		ginkgo.By("watching")
+		framework.Logf("starting watch")
+		ingWatch, err := ingClient.Watch(context.TODO(), metav1.ListOptions{ResourceVersion: ings.ResourceVersion, FieldSelector: "metadata.namespace=" + ns})
+		framework.ExpectNoError(err)
+
+		// Test cluster-wide list and watch
+		clusterIngClient := f.ClientSet.NetworkingV1beta1().Ingresses("")
+		ginkgo.By("cluster-wide listing")
+		clusterIngs, err := clusterIngClient.List(context.TODO(), metav1.ListOptions{})
```

Comment:
add a label selector filtering to limit this to `"special-label="+f.UniqueName`... this length check would fail if there were any other ingress objects anywhere in the cluster

Reply:
I would suggest setting up the listOptions as a variable at the top of the test to filter correctly, and reusing the same list options for all the list/deletecollection calls